config_option.go

```diff
@@ -0,0 +1,147 @@
+package gorm
+
+import (
+	"sync"
+	"time"
+
+	"gorm.io/gorm/clause"
+	"gorm.io/gorm/logger"
+	"gorm.io/gorm/schema"
+)
+
+// ConfigOption use functional option for gorm Config.
+type ConfigOption func(c *Config)
+
+// WithSkipDefaultTransaction enable SkipDefaultTransaction.
+func WithSkipDefaultTransaction() ConfigOption {
```

Comment:
This is totally subjective but I think I personally would prefer naming these without the "With", e.g. `SkipDefaultTransaction()`.

Reply:
Also - it might be nice for this to take an argument so you could use `WithSkipDefaultTransaction(false)` to explicitly turn it off.

Reply:
Maybe you are right, but here I want to use the option method to set the parameters by default when opening gorm DB, and then call the Apply method.